examples/MvcWeb/TheraLang/Services/ResourceService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Threading.Tasks;
+using MvcWeb.TheraLang.Entities;
+using MvcWeb.TheraLang.UnitOfWork;
+
+namespace MvcWeb.TheraLang.Services
+{
+    public class ResourceService : IResourceService
+    {
+        private readonly IUnitOfWork unitOfWork;
+
+        public ResourceService(IUnitOfWork unitOfWork)
+        {
+            this.unitOfWork = unitOfWork;
+        }
+
+        public async Task<bool> AddResource(Resource resource)
+        {
+            try
+            {
+                resource.CreatedDateUTC = DateTime.UtcNow;
+                await unitOfWork.Repository<Resource>().Add(resource);
+                await unitOfWork.SaveChangesAsync();
+                
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
```

Comment:
As I mentioned above - rewrite the logic to not return bool and throw the exception.